mountpoint-s3/src/data_cache/cache_directory.rs

```diff
@@ -16,6 +16,7 @@ use thiserror::Error;
 /// Cache directory that has been created and emptied, and will be emptied when dropped.
 #[derive(Debug)]
 pub struct ManagedCacheDir {
+    outer_path: PathBuf,
```

Comment:
If we want to add the extra complexity to this type, we should document it and make clear what are the 2 paths here, possibly with better names.